pilot/pkg/trustbundle/trustbundle_test.go

```diff
@@ -0,0 +1,135 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trustbundle
+
+import (
+	"io/ioutil"
+	"path"
+	"sort"
+	"testing"
+
+	"istio.io/istio/pkg/test/env"
+)
+
+func readCertFromFile(filename string) string {
+	csrBytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return ""
+	}
+	return string(csrBytes)
+}
+
+var testCerts map[string]string = map[string]string{
+	"MalformedCert":      "Malformed",
+	"RootCACert":         readCertFromFile(path.Join(env.IstioSrc, "samples/certs", "root-cert.pem")),
+	"NonCaCert":          readCertFromFile(path.Join(env.IstioSrc, "samples/certs", "workload-bar-cert.pem")),
+	"IntermediateCACert": readCertFromFile(path.Join(env.IstioSrc, "samples/certs", "ca-cert.pem")),
+}
+
+func TestCheckSameCerts(t *testing.T) {
```

Comment:
what about normalization? like trailing new line or not? Do those matter?

Reply:
not sure about this - think it is overkill to check for them? if there is a new trailing line, etcetera, it means that the customer has explicitly made a trustAnchor change. In that case, its okay for this function to fail and for us to push the new trustbundle to istio-agent